packages/mui-base/src/ListboxUnstyled/defaultListboxReducer.ts

```diff
@@ -286,6 +342,13 @@ export default function defaultListboxReducer<TOption>(
         ...state,
         highlightedValue: action.highlight,
       };
+    case ActionTypes.textNavigation:
+      return handleTextNavigation(
```

Comment:
> It should be a reducer's job to determine if there's a match between what was entered by the user and the options' labels. The reducer can get a string entered by the user and work out which option should be highlighted. This way, we won't have to pass isMatch and startWithCurrentOption to the reducer.

I see, let me pass the `testCriteria` (the object storing user's keyboard inputs) to the reducer, and move the `textCriteriaMatches` function to the default reducer file instead.

Reply:
> By the way - what's the intended purpose of startWithCurrentOption? The highlight should always be moved to the next option, even if the current matches.

There are use case of text navigation, the first is the example you provided. Another is as follow:

```
* Andorra
  Anguilla
  ...(a bunch of other options starts with an)
  Albania
```

user type `A`, `L`, `B`, `A`, `N` consecutively, the highlight will jump to `Albania` immediately after the `L` key, and stay at `Albania` for each key entered after.

```
  Andorra
  Anguilla
  ...(a bunch of other options starts with an)
* Albania
  Albbbb
```

If the highlight is moved to the next option even if the current matches, the entire list will need to be looped and checked once before coming back to the option that will ultimately be highlighted. I imagine this would cause performance issues when there is a high number of options?

Reply:
:+1: OK, I get it now. It seems to match the behavior of the native HTML select.

Generally, the rule seems to be that if `testCriteria` has more than one character, the currently highlighted option should be evaluated as well.